TombEngine/Objects/TR4/Trap/tr4_cog.cpp

```diff
@@ -1,64 +1,67 @@
 #include "framework.h"
-#include "tr4_cog.h"
-#include "Specific/level.h"
-#include "Game/control/control.h"
-#include "Sound/sound.h"
-#include "Game/collision/collide_room.h"
+#include "Objects/TR4/Trap/tr4_cog.h"
+
+#include "Game/animation.h"
 #include "Game/collision/collide_item.h"
+#include "Game/collision/collide_room.h"
+#include "Game/control/control.h"
 #include "Game/effects/effects.h"
-#include "Game/Lara/lara.h"
-#include "Game/animation.h"
 #include "Game/items.h"
+#include "Game/Lara/lara.h"
+#include "Sound/sound.h"
+#include "Specific/level.h"
 
 namespace TEN::Entities::Traps
 {
-    void CogControl(short itemNumber)
-    {
-        auto* item = &g_Level.Items[itemNumber];
+	void ControlCog(short itemNumber)
+	{
+		auto& item = g_Level.Items[itemNumber];
+
+		if (TriggerActive(&item))
+		{
+		   item.Status = ITEM_ACTIVE;
+			AnimateItem(&item);
+
+			if (item.TriggerFlags == 666)
```

Comment:
Looks like the original sound was the TR4_Helicopter_sound, this OCB 666 shouldn't be for the COG trap, but for the ANIMATING helicopter used in the train level.

Well, in the practice, what this does is run the object animation, play the sound in the object bone position, and then deactivate the object when the animation ends.

It works with the cog, what we need to check if if the TR4 helicopter animating has its own version of this routine.

Reply:
I pushed a commit that moved this OCB 666 and OCB 2 into the ANIMATING code.